### PR TITLE
Use new JsonMapper abstraction for serde compatibility

### PR DESCRIPTION
**Pull request by yawkat**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 projectVersion=2.0.1-SNAPSHOT
```
Comment (graemerocher) [inline gradle.properties:1]: Bump the project version to 2.1.0 snapshot

File: mqtt-core/src/main/java/io/micronaut/mqtt/serdes/JsonMqttPayloadSerDes.java [hunk 2]
```diff
@@ -43,15 +40,26 @@
...
      */
     public JsonMqttPayloadSerDes(ObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
+        this(new JacksonDatabindMapper(objectMapper));
+    }
+
+    /**
+     * Default constructor.
+     *
+     * @param jsonMapper The json mapper
```
Comment (graemerocher) [inline mqtt-core/src/main/java/io/micronaut/mqtt/serdes/JsonMqttPayloadSerDes.java:58]: add `@since 2.1.0`

**Review by graemerocher — CHANGES_REQUESTED** [review]

**Comment by yawkat** [comment]
kapt failure, do i need to update the build tools as well?

File: gradle.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-projectVersion=2.0.1-SNAPSHOT
+projectVersion=2.1.0-SNAPSHOT
 micronautDocsVersion=2.0.0.RC1
-micronautVersion=3.0.0-RC1
+micronautVersion=3.2.0
 micronautTestVersion=3.0.0-RC1
 groovyVersion=3.0.8
```
Comment (ilopmar) [inline gradle.properties:5]: Please upgrade micronaut-test and groovy

File: mqtt-core/src/main/java/io/micronaut/mqtt/serdes/JsonMqttPayloadSerDes.java [hunk 4]
```diff
@@ -43,15 +40,27 @@
...
     /**
-     * Default constructor.
+     * Legacy jackson constructor.
      *
      * @param objectMapper The jackson object mapper
+     * @deprecated Use {@link #JsonMqttPayloadSerDes(JsonMapper)} instead
      */
     public JsonMqttPayloadSerDes(ObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
+        this(new JacksonDatabindMapper(objectMapper));
```
Comment (ilopmar) [inline mqtt-core/src/main/java/io/micronaut/mqtt/serdes/JsonMqttPayloadSerDes.java:52]: I think you also need to annotate the constructor with `@Deprecated` so IDE knows about it

**Review by ilopmar — CHANGES_REQUESTED** [review]

**Comment by ilopmar** [comment]
> kapt failure, do i need to update the build tools as well?

Yes, you probably need to update to Kotlin 1.6.0 and modify the build with latest micronaut gradle internal plugins.

**Comment by yawkat** [comment]
@ilopmar ive changed the build and settings to be more like the flyway module, is this right?

**Comment by ilopmar** [comment]
@yawkat yes, I think that will work. To make sure please run `./gradlew docs` and check that the Configuration Reference docs are created.

**Comment by yawkat** [comment]
yep, they are :)

**Review by ilopmar — APPROVED** [review]

**Comment by ilopmar** [comment]
One thing I missed. You need to target core `3.3.x` branch.

**Comment by yawkat** [comment]
what do you mean?

**Comment by ilopmar** [comment]
This is core's branch that will receive the automatic "update versions PRs" when we release this module. https://github.com/micronaut-projects/micronaut-mqtt/blob/master/gradle.properties#L19

**Comment by yawkat** [comment]
i see! fixed.

**Review by graemerocher — APPROVED** [review]

**Pull request merged**